resqpy/olio/relperm.py

```diff
@@ -0,0 +1,316 @@
+               sat_cols = {'Sw', 'So'}
+               assert df.columns[0] in sat_cols and len(set(df.columns).intersection(
+                  sat_cols)) == 1, 'incorrect saturation column name and/or multiple saturation columns exist'
+               assert set(df.columns).issubset(
+                  expected_cols), f'incorrect column name(s) {set(df.columns).difference(expected_cols)}'
+            elif processed_phase_combo == {'gas', 'oil'}:
+               expected_cols = {'Sg', 'So', 'Krg', 'Kro', 'Pc'}
+               sat_cols = {'Sg', 'So'}
+               assert df.columns[0] in sat_cols and len(set(df.columns).intersection(
+                  sat_cols)) == 1, 'incorrect saturation column name and/or multiple saturation columns exist'
+               assert set(df.columns).issubset(
+                  expected_cols), f'incorrect column name(s) {set(df.columns).difference(expected_cols)}'
+            elif processed_phase_combo == {'gas', 'water'}:
+               expected_cols = {'Sg', 'Sw', 'Krg', 'Krw', 'Pc'}
+               sat_cols = {'Sg', 'Sw'}
+               assert df.columns[0] in sat_cols and len(set(df.columns).intersection(
+                  sat_cols)) == 1, 'incorrect saturation column name and/or multiple saturation columns exist'
+               assert set(df.columns).issubset(
+                  expected_cols), f'incorrect column name(s) {set(df.columns).difference(expected_cols)}'
+         elif phase_combo is None:
+            assert df.columns[0] in ['Sw', 'Sg', 'So'] and len(set(df.columns).intersection({'Sw', 'Sg',
+                                                                                             'So'})) == 1, \
+                'incorrect saturation column name and/or multiple saturation columns exist'
+            if set(df.columns).issubset({'Sw', 'So', 'Krw', 'Kro', 'Pc'}) and len(set(df.columns)) >= 3:
+               phase_combo = 'water-oil'
+            elif set(df.columns).issubset({'Sg', 'So', 'Krg', 'Kro', 'Pc'}) and len(set(df.columns)) >= 3:
+               phase_combo = 'gas-oil'
+            elif set(df.columns).issubset({'Sg', 'Sw', 'Krg', 'Krw', 'Pc'}) and len(set(df.columns)) >= 3:
+               phase_combo = 'gas-water'
+            else:
+               raise Exception('unexpected number of columns and/or column headers')
+
+         # ensure that missing capillary pressure values are stored as np.nan
+         for col in df.columns:
+            if col.capitalize() == 'Pc':
+               df[col].replace('None', np.nan, inplace = True)
+
+         # convert all values in the dataframe to numeric type
+         df_cols = df.columns
+         df[df_cols] = df[df_cols].apply(pd.to_numeric, errors = 'coerce')
+
+         # ensure that no other column besides Pc has missing values
+         cols_no_pc = [x for x in df.columns if 'pc' != x.lower()]
+         for col in cols_no_pc:
+            if col.capitalize == 'Pc':
```

Comment:
Ye of little faith!  No need to have this as well as the cols_no_pc setup: I suggest using just one or the other

Reply:
`str.capitalize` is a method, so this would need brackets `()`